src/java.base/share/classes/sun/nio/ch/SocketChannelImpl.java

```diff
@@ -700,8 +710,8 @@ private SocketAddress implBindUnix(SocketAddress local) throws IOException {
         UnixDomainSockets.checkPermission();
         UnixDomainSocketAddress usa = UnixDomainSockets.checkAddress(local);
         Path path = usa == null ? null : usa.getPath();
-        if (path != null) {
-            // implicit bind for null address
+        if (path != null && path.toString().length() > 0) {
+            // implicit bind for null address or empty path
             UnixDomainSockets.bind(fd, path);
         }
         if (usa == null || path.toString().equals("")) {
```

Comment:
Maybe you could replace `path.toString().length() > 0` with `!path.toString().isEmpty()` and `path.toString().equals("")` with `path.toString().isEmpty()`.

Reply:
Done